exonum-java-proofs/src/test/java/com/exonum/binding/storage/proofs/map/DbKeyCommonPrefixParameterizedTest.java

```diff
@@ -0,0 +1,97 @@
+package com.exonum.binding.storage.proofs.map;
+
+import static com.exonum.binding.test.Bytes.bytes;
+import static com.exonum.binding.test.Bytes.createPrefixed;
+import static com.exonum.binding.test.TestParameters.parameters;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.Collection;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class DbKeyCommonPrefixParameterizedTest {
+
+  @Parameter(0)
+  public DbKey firstKey;
+
+  @Parameter(1)
+  public DbKey secondKey;
+
+  @Parameter(2)
+  public DbKey expectedResultKey;
+
+  @Parameter(3)
+  public String description;
+
+  @Test
+  public void commonPrefixTest() {
+    DbKey actualCommonPrefixKey = firstKey.commonPrefix(secondKey);
+    assertThat(actualCommonPrefixKey, equalTo(expectedResultKey));
+  }
+
+  @Parameters(name = "{index} = {3}")
+  public static Collection<Object[]> testData() {
+    return Arrays.asList(
+        // "A | B -> C" reads "C is a common prefix of A and B"
+        parameters(
```

Comment:
I'd consider adding section comments to help avoid duplicate cases and explain what you intend to test:

// # Not a prefix: